src/trie.js

```diff
@@ -0,0 +1,53 @@
+const VALUE = Symbol('value');
```

Comment:
ported from https://github.com/theKashey/proxyequal/blob/master/src/objectTrie.js

Reply:
there is a better written, better typed and better tested version of this file - https://github.com/theKashey/search-trie/blob/master/src/word-trie.ts
We can swap internal implementation for external to make it a little more maintainable.